framework/resource_router.go

```diff
@@ -1,9 +1,110 @@
 package framework
 
-import "context"
+import (
+	"context"
+	"fmt"
 
-func DefaultResourceRouter(rs []Resource) func(ctx context.Context, obj interface{}) ([]Resource, error) {
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/micrologger/loggermeta"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/giantswarm/operatorkit/framework/context/reconciliationcanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/framework/context/updatenecessarycontext"
+)
+
+type ResourceRouterConfig struct {
+	CustomObjectVersionFunc func(obj interface{}) (string, error)
+	CtxFunc                 func(ctx context.Context, obj interface{}) (context.Context, error)
+	Logger                  micrologger.Logger
+	ResourceFunc            func(ctx context.Context, obj interface{}) ([]Resource, error)
```

Comment:
Why this has to be a function instead of `Resources []Resource`? Why it takes ctx and obj?